server/src/main/java/io/crate/metadata/functions/SignatureBinder.java

```diff
@@ -177,6 +178,10 @@ private static TypeSignature applyBoundVariables(TypeSignature typeSignature, Bo
                 new TypeSignature(baseType, parameters)
             );
         }
+        if (typeSignature instanceof IntegerLiteralTypeSignature) {
+            // Parameter, don't convert it to plain TypeSignature and return as is.
+            return typeSignature;
+        }
         return new TypeSignature(baseType, parameters);
```

Comment:
Is it ensured that the baseType doesn't change here? Or that the parameters changed to some you actually want to use?

Reply:
I think we can put this instanceof check right at the top of the method?  If the incoming TypeSignature is an integer literal then it can't be bound to a type variable at all so we can shortcut.